gradle-plugins/plugin-base/src/main/java/io/spine/gradle/SpinePlugin.java

```diff
@@ -168,25 +223,59 @@ public GradleTask applyNowTo(Project project) {
                 final String errMsg = "Project is not specified for the new Gradle task: ";
                 checkNotNull(project, errMsg + name);
 
-                if (followingTask == null && previousTask == null) {
+                if (followingTask == null
+                        && previousTask == null
+                        && previousTaskOfAllProjects == null) {
                     final String exceptionMsg =
                             "Either the previous or the following task must be set.";
                     throw new IllegalStateException(exceptionMsg);
                 }
 
                 final Task newTask = project.task(name.getValue())
                                             .doLast(action);
+                dependTask(newTask, project);
+                addTaskIO(newTask);
+                final GradleTask result = new GradleTask(name, project);
+                return result;
+            }
+
+            private void dependTask(Task task, Project project) {
                 if (previousTask != null) {
-                    newTask.dependsOn(previousTask.getValue());
+                    task.dependsOn(previousTask.getValue());
                 }
                 if (followingTask != null) {
                     final TaskContainer existingTasks = project.getTasks();
                     existingTasks.getByPath(followingTask.getValue())
-                                 .dependsOn(newTask);
+                                 .dependsOn(task);
 
                 }
-                final GradleTask result = new GradleTask(name, project);
-                return result;
+                if (previousTaskOfAllProjects != null) {
+                    final Project root = project.getRootProject();
+                    dependTaskOnAllProjects(task, root);
+                }
+            }
+
+            private void dependTaskOnAllProjects(final Task task, Project rootProject) {
+                final String prevTaskName = previousTaskOfAllProjects.getValue();
+                ProjectHierarchy.applyToAll(rootProject, new Action<Project>() {
+                    @Override
+                    public void execute(Project project) {
+                        final Task existingTask = project.getTasks().findByName(prevTaskName);
+                        if (existingTask != null) {
```

Comment:
This line is reported to be only partially tested.

Reply:
Covered both cases.